subcommands/keys/tuf_resign_root.go

```diff
@@ -28,62 +21,40 @@ Please, use a more secure way to keep your TUF root role fresh:
 - rotate the root key using "fioctl keys tuf rotate-offline-key --role=root --keys=<offline-creds.tgz>".
 `,
 	}
-	resign.Flags().BoolVarP(&dryRun, "dryrun", "", false, "Just print what the new root.json will look like and exit")
-	resign.Flags().StringVarP(&changeReason, "changelog", "m", "", "Reason for resigning root.json. Saved in root metadata for tracking change history")
+	resign.Flags().StringP("changelog", "m", "Refresh TUF root expiration by 1 year",
+		"Reason for re-signing root.json. Saved in root metadata to track change history.")
 	cmd.AddCommand(resign)
 }
 
 func doResignRoot(cmd *cobra.Command, args []string) {
+	isTufUpdatesShortcut = true
 	factory := viper.GetString("factory")
-	credsFile := args[0]
-	subcommands.AssertWritable(credsFile)
-	creds, err := GetOfflineCreds(credsFile)
-	subcommands.DieNotNil(err)
-
-	user, err := api.UserAccessDetails(factory, "self")
-	subcommands.DieNotNil(err)
-
-	root, err := api.TufRootGet(factory)
-	subcommands.DieNotNil(err)
-
-	root.Signed.Expires = time.Now().AddDate(1, 0, 0).UTC().Round(time.Second) // 1 year validity
-	root.Signed.Version += 1
+	changelog, _ := cmd.Flags().GetString("changelog")
+	keysFile := args[0]
 
-	curPk, err := findTufRootSigner(root, creds)
+	creds, err := GetOfflineCreds(keysFile)
 	subcommands.DieNotNil(err)
-	fmt.Println("= Current root:", curPk.Id)
 
-	if len(changeReason) == 0 {
-		changeReason = "resigning root.json"
-	}
-	root.Signed.Reason = &client.RootChangeReason{
-		PolisId:   user.PolisId,
-		Message:   changeReason,
-		Timestamp: time.Now(),
-	}
+	// Below the `tuf updates` subcommands are chained in a correct order.
+	// Detach from the parent, so that command calls below use correct args.
+	tufCmd.RemoveCommand(tufUpdatesCmd)
 
-	fmt.Println("= Resigning root.json")
-	removeUnusedTufKeys(root)
-	subcommands.DieNotNil(signTufRoot(root, *curPk))
+	fmt.Println("= Creating new TUF updates transaction")
+	tufUpdatesCmd.SetArgs([]string{"init", "-m", changelog})
```

Comment:
it kind of feels like you are calling "init" this way because the "init" command isn't a variable you have access to. Seems like it might be safer to make a tufInitCmd variable and call it directly? this would help spot issues at compile time, right?

Reply:
The primary reason I do it this way is because I don't want to detach every `updates` subcommand form their parent.
The `tufUpdatesCmd` serves as a facade here.

The compile time error seems a reasonable argument; but here is the thing:
- it will only check that the `init` command is still there, but not for its arguments, i.e. only 5% of the thing would be covered.

A really "static" approach would be to rewrite these shortcuts to use lower-level primitives.
But in that case we lose a huge part of code re-usability and/or cleanness which is provided by the current approach.